x/interchainstaking/keeper/keeper.go

```diff
@@ -309,6 +345,12 @@ func (k *Keeper) SetValidatorForZone(ctx sdk.Context, zone *types.Zone, data []b
 				return err
 			}
 		} else if val.Jailed && !validator.IsJailed() {
+			consAddr, err := validator.GetConsAddr()
+			if err != nil {
+				return err
+			}
+			k.DeleteValidatorAddrByConsAddr(ctx, zone.ChainId, consAddr)
+
```

Comment:
I don't think this is necessary. Once the validator exists, it will always exist, right? Not sure of the purpose of removing it here.

Reply:
The only thing to be aware of (worth noting somewhere) is that if consensus key rotation is added to cosmos, we'll need to store consensus keys for all validators, and add a check here for old vs new.

For now it's not an issue though.

Reply:
Missing this, I will update in the next commit